tasks/Task.cpp

```diff
@@ -93,8 +100,42 @@ void Task::updateHook()
     DeliveryStatus delivery_status;
     while( (delivery_status = checkDelivery(im_status, im_attempts)) != PENDING
             && im_status.first.status == PENDING)
-            im_status = std::make_pair (updateDelivery(im_status, delivery_status), base::Time::now());
+        im_status = std::make_pair (updateDelivery(im_status, delivery_status), base::Time::now());
 
+    /**
+     * Handle Position
+     */
+    bool new_pose = false;
+    base::samples::RigidBodyState pose;
+    while(_local_position.read(pose) == RTT::NewData)
+    {
+        if(pose.hasValidPosition())
+        {
+            local_position = pose;
+            new_pose = true;
+        }
+    }
+
+   while(_remote_position.read(pose) == RTT::NewData)
+    {
+        if(pose.hasValidPosition())
+        {
+            remote_position = pose;
+            new_pose = true;
+        }
+    }
+
+   if(new_pose)
+   {
+        travel_time = updateTravelTime(local_position, remote_position);
+        double prob = updateProbability(local_position, remote_position);
+        if(last_probability != prob)
+        {
+            boost::random::bernoulli_distribution<> bernoulli(prob);
+            dist = bernoulli;
```

Comment:
Can't you just do `dist = boost::random::bernoulli_distribution<> bernoulli(prob);` ?